packages/connection/src/Manager.php

```diff
@@ -28,6 +29,155 @@ class Manager implements Manager_Interface {
 	 */
 	protected $secret_callable;
 
+	/**
+	 * A copy of the raw POST data for signature verification purposes.
+	 *
+	 * @var String
+	 */
+	protected $raw_post_data;
+
+	/**
+	 * Verification data needs to be stored to properly verify everything.
+	 *
+	 * @var Object
+	 */
+	private $xmlrpc_verification = null;
+
+	/**
+	 * Creates an instance of the connection manager.
+	 */
+	public function __construct() {
+	}
+
+	/**
+	 * Initializes required listeners. This is done separately from the constructors
+	 * because some objects sometimes need to instantiate separate objects of this class.
+	 */
+	public function init() {
+
+		$is_jetpack_xmlrpc_request = $this->setup_xmlrpc_handlers(
+			$_GET,
+			$this->is_active(),
+			$this->verify_xml_rpc_signature()
+		);
+
+		// All the XMLRPC functionality has been moved into setup_xmlrpc_handlers.
+		if (
+			! $is_jetpack_xmlrpc_request
+			&& is_admin()
+			&& isset( $_POST['action'] ) // phpcs:ignore WordPress.Security.NonceVerification
```

Comment:
Definitely not a requirement, but I like adding `@todo` blocks in the docblocks of the methods when there is something that needs refactoring. It would be nice to implement a better nonce verification here at some point.

Reply:
Done in 0dd9046c77297db60bb895f86b190e148c5f9dcf